pallets/currency/src/amount.rs

```diff
@@ -1,5 +1,10 @@
 // We allow these lints because they are only caused by the 'mockable' attribute.
-#![allow(clippy::forget_non_drop, clippy::swap_ptr_to_ref, clippy::forget_ref, clippy::forget_copy)]
+#![allow(
+	clippy::forget_non_drop,
+	clippy::swap_ptr_to_ref,
+	forgetting_references,
```

Comment:
* `clippy::forget_ref`is updated to `forgetting_references`
* `clippy::forget_copy` is updated to `forgetting_copy_types` 
in the version `nightly-2024-02-09`